File: src/algorithms/sorting/novelty/spaghetti-sort/sources/spaghetti-sort.py
def spaghetti_sort(input_array: list[int]) -> list[int]:  # @step:initialize
    original_array = input_array.copy()  # @step:initialize
    array_length = len(original_array)  # @step:initialize

    remaining_strands = original_array.copy()  # @step:initialize
    sorted_result: list[int] = []  # @step:initialize

    # Repeatedly find and remove the tallest strand (maximum element)
    for _ in range(array_length):  # @step:find-tallest
        tallest_index = 0  # @step:find-tallest
        tallest_value = remaining_strands[0]  # @step:find-tallest

        # Scan all remaining strands to find the tallest
        for scan_index in range(1, len(remaining_strands)):  # @step:compare
            if remaining_strands[scan_index] > tallest_value:  # @step:compare
                tallest_index = scan_index  # @step:compare
                tallest_value = remaining_strands[scan_index]  # @step:compare

        # Remove the tallest strand and prepend it to build ascending result
        remaining_strands.pop(tallest_index)  # @step:swap
        sorted_result.insert(0, tallest_value)  # @step:swap

        # @step:mark-sorted

    return sorted_result  # @step:complete

File: src/algorithms/sorting/novelty/spaghetti-sort/sources/SpaghettiSort.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class SpaghettiSort {
    public static int[] spaghettiSort(int[] inputArray) { // @step:initialize
        int arrayLength = inputArray.length; // @step:initialize

        List<Integer> remainingStrands = new ArrayList<>(); // @step:initialize
        for (int value : inputArray) { // @step:initialize
            remainingStrands.add(value); // @step:initialize
        }

        List<Integer> sortedResult = new ArrayList<>(); // @step:initialize

        // Repeatedly find and remove the tallest strand (maximum element)
        for (int extractionPass = 0; extractionPass < arrayLength; extractionPass++) { // @step:find-tallest
            int tallestIndex = 0; // @step:find-tallest
            int tallestValue = remainingStrands.get(0); // @step:find-tallest

            // Scan all remaining strands to find the tallest
            for (int scanIndex = 1; scanIndex < remainingStrands.size(); scanIndex++) { // @step:compare
                if (remainingStrands.get(scanIndex) > tallestValue) { // @step:compare
                    tallestIndex = scanIndex; // @step:compare
                    tallestValue = remainingStrands.get(scanIndex); // @step:compare
                }
            }

            // Remove the tallest strand and prepend to build ascending result
            remainingStrands.remove(tallestIndex); // @step:swap
            sortedResult.add(0, tallestValue); // @step:swap — prepend max

            // @step:mark-sorted
        }

        return sortedResult.stream().mapToInt(Integer::intValue).toArray(); // @step:complete
    }
}
